sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/models/ShareFileProperties.java

```diff
@@ -77,6 +80,55 @@ public ShareFileProperties(final String eTag, final OffsetDateTime lastModified,
         final OffsetDateTime copyCompletionTime, final String copyStatusDescription, final String copyId,
         final String copyProgress, final String copySource, final CopyStatusType copyStatus,
         final Boolean isServerEncrypted, final FileSmbProperties smbProperties) {
+        this(eTag, lastModified, metadata,fileType, contentLength, contentType, contentMd5, contentEncoding,
```

Comment:
I think based on the discussion around deprecation we had on the cpkn thread, I'll leave this as not deprecated

Reply:
@JonathanGiles can you comment on if this would be a case for deprecation?

Reply:
It's hard to provide much useful judgement here. It comes down to whether you want users to call the API, if it provides any utility or convenience to the user, if it is behaving in an expected manner, etc. What does it mean to call this method without the lease information, and what would existing users who call this method do (would they all pass in null as you have, or would they provide non-null args which would give them a better UX).

Reply:
Customers shouldn't really ever be constructing this object themselves. It's only public because it is constructed across package boundaries. Hence, we don't want them to call either the original or the new one, it doesn't provide any utility or convenience. It does behave in the expected manner.